src/main/java/org/cactoos/text/TextOf.java

```diff
@@ -399,6 +403,210 @@ public TextOf(final LocalDate date, final DateTimeFormatter formatter) {
         );
     }
 
+    /**
+     * Formats the date using the provided formatter.
+     * @param date The date to format.
+     * @param formatter The formatter to use.
+     */
+    public TextOf(final LocalDateTime date,
+        final DateTimeFormatter formatter) {
+        super((Scalar<String>) () -> formatter.format(
+            date.atZone(ZoneId.systemDefault())
+            )
+        );
+    }
+
+    /**
+     * Formats date using ISO date time format.
+     * @param date The date to format.
+     */
+    public TextOf(final LocalDateTime date) {
+        this(date, new Iso().value());
+    }
+
+    /**
+     * Formats date using provided date time format string using default locale.
+     * @param date The date to format.
+     * @param format The format to use.
+     */
+    public TextOf(final LocalDateTime date, final String format) {
+        this(date, format, Locale.getDefault(Locale.Category.FORMAT));
+    }
+
+    /**
+     * Formats the date using the provided format string using the provided
+     * locale.
+     * @param date The date to format.
+     * @param format The format string to use.
+     * @param locale The locale to use.
+     */
+    public TextOf(final LocalDateTime date, final String format,
+        final Locale locale) {
+        this(date, DateTimeFormatter.ofPattern(format, locale));
+    }
+
+    /**
+     * Formats the milliseconds using the ISO format.
+     * @param milliseconds Milliseconds to format.
+     */
+    public TextOf(final Long milliseconds) {
```

Comment:
It is very unintuitive that `new TextOf(1234L)` is a textual date. Any reasonable user would expect it to just be a "1234L" string. And if not, why date specifically? A Long number could represent anything: distance, money, temperature, etc. I think formatting milliseconds to ISO date is a responsibility for a new class, `FormattedEpoch` or something like that. And then the no-arg `DateAsText` constructor could be moved there, as now it's lost in the migration.

This comment applies to other constructors taking milliseconds as well.